zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterUtils.java

```diff
@@ -24,18 +24,27 @@
 import java.net.InetSocketAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  *
  */
 public class RemoteInterpreterUtils {
   static Logger LOGGER = LoggerFactory.getLogger(RemoteInterpreterUtils.class);
-  public static int findRandomAvailablePortOnAllLocalInterfaces() throws IOException {
+  private static Set<Integer> portsUsing = new HashSet();
```

Comment:
Sorry for late response @bzz , For the unit test, I am thinking that I can create 100 threads (increase the possibility of port conflict) and each thread will try to find a new avaliable port. And finally I can verify that size of `portsUsing` is 100 which means no port is used by 2 threads.  What do you think ?